Content.Server/Singularity/EntitySystems/ContainmentFieldGeneratorSystem.cs

```diff
@@ -1,264 +1,387 @@
+
+    public override void Initialize()
     {
-        [Dependency] private readonly TagSystem _tags = default!;
-        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        base.Initialize();
+
+        SubscribeLocalEvent<ContainmentFieldGeneratorComponent, StartCollideEvent>(HandleGeneratorCollide);
+        SubscribeLocalEvent<ContainmentFieldGeneratorComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<ContainmentFieldGeneratorComponent, InteractHandEvent>(OnInteract);
+        SubscribeLocalEvent<ContainmentFieldGeneratorComponent, AnchorStateChangedEvent>(OnAnchorChanged);
+        SubscribeLocalEvent<ContainmentFieldGeneratorComponent, UnanchorAttemptEvent>(OnUnanchorAttempt);
+        SubscribeLocalEvent<ContainmentFieldGeneratorComponent, ComponentRemove>(OnComponentRemoved);
+    }
 
-        public override void Initialize()
-        {
-            base.Initialize();
-
-            SubscribeLocalEvent<ContainmentFieldGeneratorComponent, ComponentRemove>(OnComponentRemoved);
-            SubscribeLocalEvent<ContainmentFieldComponent, StartCollideEvent>(HandleFieldCollide);
-            SubscribeLocalEvent<ContainmentFieldGeneratorComponent, StartCollideEvent>(HandleGeneratorCollide);
-            SubscribeLocalEvent<ParticleProjectileComponent, StartCollideEvent>(HandleParticleCollide);
-            SubscribeLocalEvent<ContainmentFieldGeneratorComponent, AnchorStateChangedEvent>(OnAnchorChanged);
-            SubscribeLocalEvent<ContainmentFieldGeneratorComponent, UnanchorAttemptEvent>(OnUnanchorAttempt);
-            SubscribeLocalEvent<ContainmentFieldGeneratorComponent, InteractHandEvent>(OnInteract);
-        }
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
 
-        private void OnInteract(EntityUid uid, ContainmentFieldGeneratorComponent component, InteractHandEvent args)
+        foreach (var generator in EntityQuery<ContainmentFieldGeneratorComponent>())
         {
-            if (args.Handled)
+            if (generator.PowerBuffer <= 0) //don't drain power if there's no power, or if it's somehow less than 0.
                 return;
 
-            if (TryComp(component.Owner, out TransformComponent? transformComp) && transformComp.Anchored)
+            generator.Accumulator += frameTime;
+
+            if (generator.Accumulator >= generator.Threshold)
             {
-                if (!component.Enabled)
-                    TurnOn(component);
-                else if (component.Enabled && component.IsConnected)
-                {
-                    _popupSystem.PopupEntity(Loc.GetString("comp-containment-anchor-warning"), args.User, Filter.Entities(args.User));
-                    return;
-                }
-                else
-                    TurnOff(component);
+                LosePower(generator.PowerLoss, generator);
+                generator.Accumulator = 0f;
```

Comment:
```suggestion
                generator.Accumulator -= generator.Threshold;
```

This way long-term it should be more similar.